scapy/contrib/automotive/gm/gmlan_scanner.py

```diff
@@ -0,0 +1,688 @@
+        if global_configuration.unittest:
+            return
+
+        if cast(ServiceEnumerator, self)._retry_pkt[state] is not None:
+            # this is a retry execute. Wait much longer than usual because
+            # a required time delay not expired could have been received
+            # on the previous attempt
+            time.sleep(11)
+        else:
+            time.sleep(1)
+
+    def _evaluate_response(self, state, request, response, **kwargs):
+        # type: (EcuState, Packet, Optional[Packet], Optional[Dict[str, Any]]) -> bool  # noqa: E501
+        if super(GMLAN_SAEnumerator, self)._evaluate_response(
+                state, request, response, **kwargs):
+            return True
+
+        if response is None:
+            # Nothing to evaluate, return and continue execute
+            return False
+
+        if response.service == 0x7f and \
+                self._get_negative_response_code(response) in [0x22, 0x37]:
+            # requiredTimeDelayNotExpired or requestSequenceError
+            if cast(ServiceEnumerator, self)._retry_pkt[state] is None:
+                # This was no retry since the retry_pkt is None
+                self._retry_pkt[state] = request
+                log_interactive.debug(
+                    "[-] Exit execute. %s. Retry next time.",
+                    self._get_negative_response_label(response))
+                return True
+            else:
+                # This was an unsuccessful retry, continue execute
+                self._retry_pkt[state] = None
+                log_interactive.warning(
+                    "[-] Unsuccessful retry. %s!",
+                    self._get_negative_response_label(response))
+                return False
+        else:
+            self._retry_pkt[state] = None
+
+        if response.service == 0x67 and response.subfunction % 2 == 1:
+            log_interactive.debug("[i] Seed received. Leave scan to try a key")
+            return True
+
+        return False
+
+    @staticmethod
+    def get_seed_pkt(sock, level=1):
+        # type: (_SocketUnion, int) -> Optional[Packet]
+        req = GMLAN() / GMLAN_SA(subfunction=level)
+        seed = None
+        for t in range(10):
+            seed = sock.sr1(req, timeout=5, verbose=False)
+            if seed is None or \
+                    (seed.service == 0x7f and
+                     GMLAN_Enumerator._get_negative_response_code(seed) != 0x37):  # noqa: E501
+                log_interactive.info(
+                    "Security access no seed! NR: %s", repr(seed))
+                return None
+
+            if seed.service == 0x7f and \
+                    GMLAN_Enumerator._get_negative_response_code(seed) == 0x37:
+                log_interactive.info("Security access retry to get seed")
+                time.sleep(10)
+                continue
+
+            if t == 9:
+                return None
```

Comment:
You can use the 
```
for ...:

else:
```
syntax instead

Reply:
oh wow! pretty cool syntax. Didn't know it existed. Thanks for showing!